source/game.production.productionconcept.bmx

```diff
@@ -1116,11 +1122,55 @@ endrem
 			else
 				castFit = 0
 			endif
+			castFit:* _GetCastConsistencyMod()
 		endif
 
 		return castFit
 	End Method
 
+	'job-dependent weight for a calculated value
+	'in particular lower weight (for now always 1) for supporting actor
+	'weightType 1 for castFit, 2 for castExperience, 3 for fame
+	Method _GetJobWeight:Int(job:Int, weightType:Int)
+		'genre could also be taken into account
+		'e.g. script writer not as important for events
+		'but the database will have few "nonsense-job"
+
+		If TVTPersonJob.DIRECTOR & job Then Return 2
+		If TVTPersonJob.ACTOR & job Then Return 2
+
+		If weightType = 1 'for cast fit
+			If TVTPersonJob.SCRIPTWRITER & job Then Return 2
+			If TVTPersonJob.HOST & job Then Return 2
+		ElseIf weightType = 2 'for cast experience
+			If TVTPersonJob.SCRIPTWRITER & job Then Return 2
+			If TVTPersonJob.MUSICIAN & job Then Return 2
+			If TVTPersonJob.REPORTER & job Then Return 2
+			If TVTPersonJob.HOST & job Then Return 2
+		ElseIf weightType = 3 'for cast fame
+			If TVTPersonJob.GUEST & job Then Return 3
+			If TVTPersonJob.MUSICIAN & job Then Return 2
+		Else
+			throw "TProductionConcept: illegal job weight type"
+		EndIf
+		Return 1
+	End Method
+
+
+
+	'TODO cast consistency
+	Method _GetCastConsistencyMod:Float()
+		'for series consistent cast (actors in particular if roles are involved)
+		'for multi-productions host consistency and varying guests!!
```

Comment:
dunno if we should add a flag which can disable the requirement of "consistency" ... 

Just consider a show where the host is an other one each time (as part of the show "design") or something like Dr. Who ... where the doctor can change.
Or a way more... realistic thing: an adult show where the hosts just must be pretty adult who gets rid of their clothes.

Yes, rather rare - but if we allow "disabling" of such mods with flags we could model such stuff more easily

Reply:
Doctor Who - Konsistenz innerhalb einer Staffel der Serie wäre durchaus gewünscht und über mehrere Serien hinweg, würde ich gar nichts machen wollen (Franchise...).
Für das andere Beispiel reichen vielleicht TargetGroupAttractivity... Der Bonus/Malus für Konsistenz/Inkonsistenz muss ja nicht riesig sein und wäre vielleicht nicht mal der Fokus des ersten großen Refactorings für die Eigenproduktionen.

Reply:
Ich wollte es nur erwaehnt haben -- im Hinterkopf behalten und gut :D